modules/networking/cloudrun.tf

```diff
@@ -0,0 +1,30 @@
+// This file grants the Cloud Run service access to shared network resources, even if they exist in a different project.
+
+data "google_project" "cloudrun_service_project" {
+  // Prefer service_project_id, but fall back to project_id.
+  // Technically we don't need this if the service_project_id == project_id,
+  // but it's safe to always do this, since we'd be granting permissions the service account
+  // already has.
+  project_id = var.service_project_id != null ? var.service_project_id : var.project_id
+}
+
+// Grant Cloud Run access to view networks.
+// See https://cloud.google.com/run/docs/configuring/shared-vpc-direct-vpc#set_up_iam_permissions
+resource "google_project_iam_member" "cloudrun_service_network_user" {
+  project = var.project_id
+  member  = "serviceAccount:service-${data.google_project.cloudrun_service_project.number}@serverless-robot-prod.iam.gserviceaccount.com"
+  role    = "roles/compute.networkViewer"
+}
+
+// Grant Cloud Run the service access to use the networks.
+// See https://cloud.google.com/run/docs/configuring/shared-vpc-direct-vpc#set_up_iam_permissions
+resource "google_compute_subnetwork_iam_member" "subnet_network_user" {
+  for_each = resource.google_compute_subnetwork.regional
+
+  // If not set, provider project should be used.
+  project    = var.project_id
+  region     = each.value.region
+  subnetwork = each.value.name
+  role       = "roles/compute.networkUser"
+  member     = "serviceAccount:service-${data.google_project.cloudrun_service_project.number}@serverless-robot-prod.iam.gserviceaccount.com"
+}
```

Comment:
I don't think this belongs here.  It feels like we need this in exactly the contexts where we aren't using the networking module?

Reply:
My understanding is the networking module sets up all the resources we need for a VPC network. In that sense, what might make more sense is to move out the network creation out of the module and always require users to pass in the network id so we can configure everything else around it. To me these settings are more in line with the Cloud Run DNS + peering setup, which IIUC is also a one time per-environment configuration we need to make (instead of per-service).

Re breaking changes - IIUC I think this is an additive change because it's using iam_member instead of iam_binding? If the network project matches the service project, then this is granting permissions the Cloud Run service account already has, which is safe to do.
I know you had concerns around if Cloud Run services are removed we also yanking the permissions, but if I'm understanding the intended usage of the network module those won't apply here since there should only be one for each environment?
lmk if there's something I'm missing though 🤔

Reply:
I commented on the piece that is breaking below.

My point around the multi-project stuff (including this diff) is that it's irrelevant to the single project case this is designed for and we are bolting parts onto this and overloading it's purpose to meet the needs of our strange downstream setup.

Rather than scope creeping this module and trying to call it downstream, I think that we should just be trying to mirror it's functionality within our strange multi-project setup.